libcxx/test/std/algorithms/alg.nonmodifying/alg.fold/fold_left/valid.pass.cpp

```diff
@@ -0,0 +1,93 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// <algorithm>
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17, c++20
+
+// template<input_iterator I, sentinel_for<I> S, class T,
```

Comment:
We should also test that the algorithm SFINAEs away when given types not satisfying the constraints, unless you think that's redundant?

Reply:
Resolving, as that's handled by the (incoming) `requirements.compile.pass.cpp`.